### PR TITLE
feat(policy-templates): add APIGatewayFullAccessPolicy

### PR DESCRIPTION
**Pull request by meleksomai**
*Issue #, if available:*

*Description of changes:*
Added a policy template APIGatewayFullAccessPolicy that grants full access to API Gateway resources.

*Description of how you validated changes:*
I ran make `pr`.

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [x] Update documentation  _(no changes needed)_
- [x] Verify transformed template deploys and application functions as expected
- [x] Add/update example to `examples/2016-10-31`
- [x] Add Unit Testing `tests/translator/input/all_policy_templates.yaml`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/serverless-application-model/pull/1454?src=pr&el=h1) Report
> Merging [#1454](https://codecov.io/gh/awslabs/serverless-application-model/pull/1454?src=pr&el=desc) into [develop](https://codecov.io/gh/awslabs/serverless-application-model/commit/ac5b0647031b2f45c986ecdefbcf910778319b70&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/awslabs/serverless-application-model/pull/1454?src=pr&el=tree)

```diff
@@           Coverage Diff            @@
##           develop    #1454   +/-   ##
========================================
  Coverage    94.36%   94.36%           
========================================
  Files           78       78           
  Lines         4685     4685           
  Branches       937      937           
========================================
  Hits          4421     4421           
  Misses         121      121           
  Partials       143      143           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/awslabs/serverless-application-model/pull/1454?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/awslabs/serverless-application-model/pull/1454?src=pr&el=footer). Last update [ac5b064...833d893](https://codecov.io/gh/awslabs/serverless-application-model/pull/1454?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by ShreyaGangishetty** [comment]
@meleksomai Thank you for contributing this!
Could you also please add the unit tests for `APIGatewayFullAccessPolicy` [here](https://github.com/awslabs/serverless-application-model/blob/develop/tests/translator/input/all_policy_templates.yaml). Otherwise this PR looks good to me

File: samtranslator/policy_templates_data/policy_templates.json [hunk 1]
```diff
@@ -1930,6 +1930,25 @@
           }
         ]
       }
-    }
+    },
+    "ApiGatewayFullPolicy": {
```
Comment (ShreyaGangishetty) [inline samtranslator/policy_templates_data/policy_templates.json]: There is a typo in the access policy name: `ApiGatewayFullAccessPolicy`

Comment (meleksomai) [reply samtranslator/policy_templates_data/policy_templates.json]: I agree. ApiGatewayFullAccessPolicy is more appropriate.

**Comment by meleksomai** [comment]
I have updated the name of the policy and included it to the unit testing.

Thanks for the feedback.

**Comment by meleksomai** [comment]
@ShreyaGangishetty , 

I am not sure why I have the error `UnboundLocalError: local variable 'patching' referenced before assignment`.  The package `parameterized` makes it very difficult to identify the error.

Thanks

**Comment by ShreyaGangishetty** [comment]
@meleksomai I will check the error and update the PR if that works!

**Comment by ShreyaGangishetty** [comment]
@meleksomai I have fixed the tests but i don't have write access to you repository. Here is the PR i created: https://github.com/inception-labs/serverless-application-model/pull/1

**Comment by awood45** [comment]
Circling back on this, generally policy templates make more sense where a managed policy cannot accomplish the same thing, for example when you need to scope on resources, or if there are particular common compute access patterns. This seems to be identical to the `arn:aws:iam::aws:policy/AmazonAPIGatewayAdministrator` managed policy.

If there's a particular scoping that's important here that distinguishes, can reopen. Otherwise, closing and recommending use of `arn:aws:iam::aws:policy/AmazonAPIGatewayAdministrator`

**Pull request closed without merging**